### DISCUSSION TITLE
Pimcore Portal Engine Installation error

### DISCUSSION THREAD

**Discussion by ManoDharmalingam** [pimcore/pimcore#14900] [Q&A / Support] [answered]
Hi ,
We have procured Pimcore's Enterprise version and have attempted to install the Pimcore portal engine following the instructions below,
**https://pimcore.com/docs/platform/Portal_Engine/**
however, during the portal engine installation, we encountered the following error.
**[ERROR] {"error":{"root_cause":[{"type":"security_exception","reason":"missing authentication credentials for REST
         request [/_alias/portalengine_filterdefinition]","header":{"WWW-Authenticate":["Basic realm=\"security\"
         charset=\"UTF-8\"","ApiKey"]}}],"type":"security_exception","reason":"missing authentication credentials for
         REST request [/_alias/portalengine_filterdefinition]","header":{"WWW-Authenticate":["Basic realm=\"security\"
         charset=\"UTF-8\"","ApiKey"]}},"status":401}**

We would appreciate your assistance in resolving this issue.
**The following is our system configuration for the setup of the portal engine
Pimcore version: 10.5.20
Elastic search: Elastic 8.7**

Regards,
Mano Dharmalingam

**Comment by fashxp** [comment]
Hello,
when exactly that error occurs? 

BR

**Reply by ManoDharmalingam** [reply]
Thanks  for your response, please find the steps that i have done to recreate the issue
Step1: I have run the command  **"./bin/console pimcore:bundle:enable PimcorePortalEngineBundle"** and it did not show any error
Step2:  when I run the command **"/bin/console pimcore:bundle:install PimcorePortalEngineBundle"** am receiving the following error
**[ERROR] {"error":{"root_cause":[{"type":"security_exception","reason":"missing authentication credentials for REST
         request [/_alias/portalengine_filterdefinition]","header":{"WWW-Authenticate":["Basic realm="security"
         charset="UTF-8"","ApiKey"]}}],"type":"security_exception","reason":"missing authentication credentials for
         REST request [/_alias/portalengine_filterdefinition]","header":{"WWW-Authenticate":["Basic realm="security"
         charset="UTF-8"","ApiKey"]}},"status":401}**

Please let me know if you need more details

Regards,
Mano Dharmalingam

**Comment by fashxp** [comment]
please see https://pimcore.com/docs/portal-engine/current/Installation/index.html#page_Installation

![image](https://user-images.githubusercontent.com/8792145/231503807-8b217b14-f525-41ec-85f4-976d94465dfb.png)

how to setup basic auth see https://pimcore.com/docs/portal-engine/current/Installation/index.html#page_Setup-Basic-Auth

**Reply by arulselvan008** [reply]
Hi,

I created a Basic auth for Elasticsearch.

File config.yaml -->
parameters:
  pimcore_portal_engine.elasticsearch.host: 'elastic:mBqvoFk9wMWxgoAbP5GD@localhost:9200'
  pimcore_portal_engine.elasticsearch.index_prefix: 'portalengine_'
  pimcore_portal_engine:
     index_service:
          es_client_params:
              connection_params:
                  client:
                      curl:
                          !php/const CURLOPT_HTTPAUTH: !php/const CURLAUTH_BASIC
                          !php/const CURLOPT_USERPWD: "elastic:mBqvoFk9wMWxgoAbP5GD"
                         
While trying to install the portal engine bundle again, I got an error.--> Warning: Undefined array key "acknowledged"

**Comment by ManoDharmalingam** [comment]
Hi fashxp,

Still, i could not fix the authentication issue, Is there any way that we could reach out to fix that issue?

Regards,
Mano Dharmalingam.

**Reply by fashxp** [reply, marked as answer]
What I just noticed ... Pimcore 10.5 and compatible Portal Engine Versions are not compatible to Elastic search 8 ... that might be the reason for your problems. 

If you need further assistance, Please contact your contact person at Pimcore for professional support.

**Reply by ManoDharmalingam** [reply]
Thank you, fashxp. After removing Elastic Search 8 and installing Elastic Search 7.10.2, we can install Pimcore portal engine now. This solution has been marked as an answer. I need another suggestion from your side regarding the Pimcore portal engine configuration error that I intend to create as a separate discussion.
